docs/dev/lwt-implementation.md

```diff
@@ -0,0 +1,166 @@
+
+3. create mutation based on the read data from 2
+
+4. create a proposal based on the ballot from 1 and mutation from 3
+
+5. try to accept the proposal, learn if accepted, sleep/retry if failed
+
+`begin_and_repair_paxos` is similar in purpose to the prepare phase from the Simple Paxos above, but its logic and invariants are very different.
+
+It runs a loop, on each iteration it chooses a new wall-clock based ballot and tries to find a quorum of replicas that satisfy the following three invariant:
+* no replica has promised a greater or equal ballot
+* there were no previously accepted proposals on any replicas OR such an accepted proposal is greater than the most recently committed ballot
+* there are no not-pruned commited proposals OR all replicas agree on its ballot
+
+* It retries with a newer ballot as soon as it gets a first prepare reject, not trying to collect a (still possible) quorum of prepare. This catches the situation when some other proposal with higher ballot is already under way, so we give up early to not compete with it.
+
+The prepare_summary structure:
+	most_recent_commit
+		is populated with the most recent value from the most_recent_commit column from the replicas, participated in the prepare round. If prune finished for this partition key, then this column will not be set on the replicas and in summary.most_recent_commit field.
+
+	most_recent_proposal
+		is populated with the most recent value from the accept_proposal column from the replicas, participated in the prepare round.
+
+## Questions:
+
+* Why we use db::consistency_level::ANY in repair code in begin_and_repair_paxos? Doesn't this mean that we only wait for one replica to confirm? This will be a different replica from the replica where we get the most recent commit in the first place, so in fact we get two confirmations, but it's still not enough for a quorum on a larger replica sets. It's not guaranteed that the quorum read for CAS conditions in storage_proxy::cas would return the latest value.
+
+* Why we run "Finishing incomplete paxos round" code in begin_and_repair_paxos based on most_recent_commit? Doesn't this mean that we'll run it practically all the time? (summary.most_recent_commit is not set since prune clears it and most_recent_proposal is set since there were some accepted proposals for the key).
+
+* How liveness is guaranteed in the condition for "Finishing incomplete paxos round"? What if prune will be able to clear the most_recent_commit column on all nodes before we reach the next iteration in the top-level loop in begin_and_repair_paxos?
+
+* Why don't we update the `min_timestamp_micros_to_use` in `begin_and_repair_paxos` when `summary.promised` is false? This means that the call `summary.update_most_recent_promised_ballot(response);` when a replica rejects a proposal has no any practical effects. Why do we sleep for up to 50 millis? This is a huge amount of time, purely wasted.
```

Comment:
Again. Do not understand. The sleep is to back off in case parallel operations are detected.